tests/languages/pygrep_test.py

```diff
@@ -30,6 +33,40 @@ def test_main(cap_out, pattern, expected_retcode, expected_out):
     assert out == expected_out
 
 
+@pytest.mark.usefixtures('some_files')
+@pytest.mark.parametrize(
+    ('pattern', 'expected_retcode', 'expected_out'),
+    (
+        ('baz', 1, 'f4\nf5\nf6\n'),
+        ('pattern', 0, ''),
+        ('foo', 1, 'f5\n'),
+        ('pattern\nbar', 1, 'f4\nf5\nf6\n'),
+    ),
+)
+def test_negate_by_line(cap_out, pattern, expected_retcode, expected_out):
+    ret = pygrep.main((pattern, 'f4', 'f5', 'f6', '--negate'))
+    out = cap_out.get()
+    assert ret == expected_retcode
+    assert out == expected_out
+
+
+@pytest.mark.usefixtures('some_files')
+@pytest.mark.parametrize(
+    ('pattern', 'expected_retcode', 'expected_out'),
+    (
+        ('baz', 1, 'f4\nf5\nf6\n'),
+        ('pattern', 0, ''),
+        ('foo', 1, 'f5\n'),
+        ('pattern\nbar', 0, ''),
+    ),
+)
+def test_negate_by_file(cap_out, pattern, expected_retcode, expected_out):
+    ret = pygrep.main(('--multiline', pattern, 'f4', 'f5', 'f6', '--negate'))
+    out = cap_out.get()
+    assert ret == expected_retcode
+    assert out == expected_out
```

Comment:
these tests might be easier to understand without parametrize -- 4 separate tests showing positive / negative for each case?  up to you